cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -153,6 +150,8 @@ protected AbstractDistributedProgramRunner(TwillRunner twillRunner, LocationFact
     this.hConf = hConf;
     this.cConf = cConf;
     this.eventHandler = createEventHandler(cConf);
+    this.baseInjector = createBaseInjector(cConf, hConf);
+    this.secureStoreUpdater = baseInjector.getInstance(TokenSecureStoreUpdater.class);
```

Comment:
Why create a new Injector? The secure store updater instance should be passed through constructor. The sub-class will pass it in through guice injection (All sub-classes constructors already annotated with `@Inject`).